DOC:Use of "Yields" for documentation of DataFrame.iteritems()

sameshl opened the pull request:
changed `Returns` to `Yields`

closes #27860

sameshl committed:
DOC:Use of "Yields" for documentation of DataFrame.iteritems()

changed `Returns` to `Yields`

closes #27860

gfyoung commented:
@datapythonista @WillAyd : The fix looks good, but the documentation checks seem to be failing.

jorisvandenbossche commented:
See the also the discussion on the issue: https://github.com/pandas-dev/pandas/issues/27860. Someone will need to understand why the docstring validation sees iteritems and items differently

jorisvandenbossche commented:
Ah, it is because `iteritems` does `return self.items()` to do the same, therefore the docstring validation sees 'return' and thinks it needs to return something. See logic in 

https://github.com/pandas-dev/pandas/blob/master/scripts/validate_docstrings.py#L529-L562

That's of course a bit brittle. @datapythonista I suppose the easiest here is to ignore that rule? 
But currently we can only fully ignore a rule or not in code_checks ?

datapythonista commented:
Since we're Python 3 only now, my preference is to use `yield from self.items()` instead of `return self.items()`, and that should fix the doc validation I think.

sameshl committed:
MNT:Changed the return method of DataFrame.iteritems

changed `return self.items`` --> `yield from self.items()`
for better readability and fix doc string failures

jorisvandenbossche commented on pandas/core/frame.py:
```diff
@@ -825,9 +825,9 @@ def items(self):
                 yield k, self._ixs(i, axis=1)
 
-    @Appender(_shared_docs["items"] % "Returns\n        -------")
+    @Appender(_shared_docs["items"] % "Yields\n        -------")
```
@sameshl you can actually remove this `% Yields` now, and directly edit this in the docstring above. This was only done so that `items` and `iteritems` could have a different title in the docstring (returns vs yields)

sameshl replied:
Ahh. Nice suggestion @jorisvandenbossche. Will make the required change.

jorisvandenbossche commented:
thanks @datapythonista that indeed seems a good idea

sameshl committed:
edit docstring of iteritems

removed placeholder in shared docstring of iteritems and directly changed the
main docstring so both of them read `Yields` rather than `Return`.

datapythonista reviewed (approved):
looks good, thanks @sameshl 

I think it'd probably be cleaner if `def items` had the docstring directly (just a normal docstring), and `def iteritems` uses `@Appender(DataFrame.items)` (I think that's what we use in these cases).

Didn't check in detail if that makes sense for sure, and happy to do that in a follow up PR.

jorisvandenbossche commented:
Thanks @sameshl !

sameshl commented:
Glad to help! @jorisvandenbossche
